arcgis-ios-sdk-samples/Layers/ArcGIS vector tiled layer (custom style)/VectorTileCustomStyleViewController.swift

```diff
@@ -17,40 +17,122 @@ import UIKit
+        let portalItem = AGSPortalItem(url: vectorTiledLayerURL)!
+        // Create a task to export the custom style resources.
+        let task = AGSExportVectorTilesTask(portalItem: portalItem)
+        // Get the AGSExportVectorTilesJob.
+        exportVectorTilesJob = task.exportStyleResourceCacheJob(withDownloadDirectory: temporaryURL)
+        // Start the job.
+        exportVectorTilesJob?.start(
+            statusHandler: { (status: AGSJobStatus) in
+                SVProgressHUD.show(withStatus: status.statusString())
+            }, completion: { [weak self] (result, error) in
+                SVProgressHUD.dismiss()
+                guard let self = self, let map = self.mapView.map else { return }
+            if let result = result {
+                // Get the item resource cache from the result.
+                let itemresourceCahce = result.itemResourceCache
+                switch itemID {
+                case "e01262ef2a4f4d91897d9bbd3a9b1075":
+                    // Create a vector tiled layer with the vector tiled cache and the item resource cache.
+                    let dayVectorTiledLayer = AGSArcGISVectorTiledLayer(vectorTileCache: vectorTileCache, itemResourceCache: itemresourceCahce)
+                    map.basemap = AGSBasemap(baseLayer: dayVectorTiledLayer)
+                    self.dayVectorTiledLayer = dayVectorTiledLayer
+                case "ce8a34e5d4ca4fa193a097511daa8855":
+                    // Create a vector tiled layer with the vector tiled cache and the item resource cache.
+                    let nightVectorTiledLayer = AGSArcGISVectorTiledLayer(vectorTileCache: vectorTileCache, itemResourceCache: itemresourceCahce)
+                    map.basemap = AGSBasemap(baseLayer: nightVectorTiledLayer)
+                    self.nightVectorTiledLayer = nightVectorTiledLayer
+                default:
+                    fatalError("Unidentified item ID")
+                }
+            } else if let error = error {
+                // Handle errors.
+                self.presentAlert(error: error)
+            }
+            }
+        )
+        try? FileManager.default.removeItem(at: temporaryURL)
     }
     
     private func showSelectedItem(_ itemID: String) {
+        guard let map = mapView.map else { return }
+        shownItemID = itemID
+        // Set the viewpoint to display Dodge City, KS.
+        let point = AGSPoint(x: -100.01766, y: 37.76528, spatialReference: .wgs84())
+        mapView.setViewpoint(AGSViewpoint(center: point, scale: 40000))
+        // Get the vector tiled layer URL.
         let vectorTiledLayerURL = URL(string: "https://arcgisruntime.maps.arcgis.com/home/item.html?id=\(itemID)")!
-        let vectorTiledLayer = AGSArcGISVectorTiledLayer(url: vectorTiledLayerURL)
-        mapView.map?.basemap = AGSBasemap(baseLayer: vectorTiledLayer)
+        switch itemID {
+        case "e01262ef2a4f4d91897d9bbd3a9b1075":
+            if dayVectorTiledLayer == nil {
```

Comment:
I actually tried to change` itemIDs` into key value pairs since dictionaries don't always load in the same order. But I can't store the offline `AGSArcGISVectorTiledLayer`s in the key value pairs because they must be loaded using an `AGSExportVectorTilesJob`; the result of which can't be assigned on the main thread. Please let me know if I can clarify that. It's a little difficult to explain 😅